src/main/scala/com/microsoft/hyperspace/index/factories.scala

```diff
@@ -20,31 +20,32 @@ import org.apache.hadoop.conf.Configuration
 import org.apache.hadoop.fs.{FileSystem, Path}
 
 trait IndexLogManagerFactory {
-  def create(indexPath: Path): IndexLogManager
+  def create(indexPath: Path, configuration: Configuration = new Configuration): IndexLogManager
 }
 
 object IndexLogManagerFactoryImpl extends IndexLogManagerFactory {
-  override def create(indexPath: Path): IndexLogManager = {
-    new IndexLogManagerImpl(indexPath)
+  override def create(indexPath: Path,
+      configuration: Configuration = new Configuration): IndexLogManager = {
+    new IndexLogManagerImpl(indexPath, configuration)
   }
 }
 
 trait IndexDataManagerFactory {
-  def create(indexPath: Path): IndexDataManager
+  def create(indexPath: Path, configuration: Configuration = new Configuration): IndexDataManager
 }
 
 object IndexDataManagerFactoryImpl extends IndexDataManagerFactory {
-  override def create(indexPath: Path): IndexDataManager = {
-    new IndexDataManagerImpl(indexPath)
+  override def create(indexPath: Path,
+      configuration: Configuration = new Configuration): IndexDataManager = {
+    new IndexDataManagerImpl(indexPath, configuration)
   }
 }
 
 trait FileSystemFactory {
-  def create(path: Path): FileSystem
+  def create(path: Path, configuration: Configuration = new Configuration): FileSystem
 }
 
 object FileSystemFactoryImpl extends FileSystemFactory {
-  private val configuration = new Configuration
-
-  override def create(path: Path): FileSystem = path.getFileSystem(configuration)
+  override def create(path: Path, configuration: Configuration = new Configuration): FileSystem =
```

Comment:
Looks like these `*Impl` classes will always be created with default configurations right (since I don't see changes when `CachingIndexCollectionManager` or `IndexCollectionManager` is created). Does it make sense to use the same `Configuration` setting everywhere?

Reply:
👍 You're right! I missed the places where the `create` methods of `IndexLogManagerFactory` & `IndexDataManagerFactory`.